### PR TITLE
fix: add `use` to types

### PR DESCRIPTION
**Pull request by SimenB**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

Depends on https://github.com/fastify/fastify/pull/2614

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
   },
   "homepage": "https://github.com/fastify/fastify-express#readme",
   "devDependencies": {
+    "@types/express": "*",
```
Comment (Ethan-Arrowood) [inline package.json:28]: Can you add a note to the readme about users needing to install @types/express in order to use this module with typescript?

Comment (SimenB) [reply package.json:28]: sure! should I add an optional peer dep as well?

Comment (Ethan-Arrowood) [reply package.json:28]: We don't really use peerDeps in fastify - we've settled on just listing typescript deps in readmes/docs cause usually they are quite sparse

Comment (SimenB) [reply package.json:28]: Cool, done 🙂 (did the same for the `middie` PR)

**Review by Ethan-Arrowood — APPROVED** [review]
lgtm :) great work!

**Review by fox1t — APPROVED** [review]
LGTM

File: README.md [hunk 2]
```diff
@@ -144,6 +144,10 @@ async function subsystem (fastify, opts) {
 }
 ```
 
+## TypeScript support
+
+To use this module with TypeScript, make sure to install `@types/express`.
```
Comment (SimenB) [inline README.md:149]: this is already true on master, might wanna land the readme changes while waiting on a decision whether or not this should land at all?

**Comment by SimenB** [comment]
@mcollina rebased this to trigger CI with fastify@3.7. Tests pass locally at least 🙂

**Pull request merged**